pkg/vsphere/disk/disk_manager.go

```diff
@@ -223,6 +223,20 @@ func (m *Manager) Create(ctx context.Context, newDiskURI string,
 //	return nil
 // }
 
+func (m *Manager) Attach(ctx context.Context, spec *types.VirtualDisk) error {
+	machineSpec := configureDeviceSpec(ctx, *m.vm, types.VirtualDeviceConfigSpecOperationAdd, types.VirtualDeviceConfigSpecFileOperationCreate, spec)
+
+	_, err := tasks.WaitAndRetryForResult(ctx, func(ctx context.Context) (tasks.ResultWaiter, error) {
```

Comment:
The PR itself is out against a p0 issue blocking volume ls tests. Which means that the best way to test and make for sure that function is working as expected is to see if the fault we are seeing is now blocking the tests.